zcash_client_sqlite/src/wallet.rs

```diff
@@ -536,6 +536,51 @@ pub(crate) fn block_height_extrema(
     })
 }
 
+pub(crate) fn fully_scanned_height(
+    conn: &rusqlite::Connection,
+) -> Result<Option<(BlockHeight, CommitmentTreeMeta)>, SqliteClientError> {
+    let res_opt = conn
+        .query_row(
+            "SELECT height, sapling_commitment_tree_size, sapling_tree
+            FROM blocks
+            ORDER BY height DESC
+            LIMIT 1",
```

Comment:
This isn't a bug _yet_. It will be a bug once out-of-order scan range suggestion is implemented, but at present this is still correct for the linear scanning. This will be fixed in a follow-up PR; it doesn't make sense to fix it here because it would require querying the entire blocks table to look for discontinuities.